rustls/src/client/hs.rs

```diff
@@ -68,7 +68,9 @@ fn find_session(
             None
         })
         .and_then(|resuming| {
-            let retrieved = persist::Retrieved::new(resuming, UnixTime::now());
+            let now = config.get_current_time().ok()?;
```

Comment:
I think matching the err result here and `debug!` logging would be helpful. If the time provider is broken session resumption would silently return `None` for all sessions.